### PR TITLE
feat(config): improve monorepo names

### LINKED ISSUE #10210: Use better monorepo names in `monorepo.ts`
**What would you like Renovate to be able to do?**

I made a PR to add a new monorepo to the list, and saw that @viceice changed the name of the repo from `fontsource` -> `Fontsource`. But other names in this list use lowercase, so we're not consistent with capitalization of the mono repo names.

I'd like to use proper names and make things consistent again, but that's a potentially breaking change. 😄 

**Did you already have any implementation ideas?**

I'd like some feedback from the Renovate team on what the proper names are, and if we want this change at all. 😄 

If I know what we want, and if we want it, I'll then make a draft PR with the `breaking` label.

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- General cleanup of monorepo names

## Context:

Found a lot of inconsistencies, so I'm labeling this a breaking change. 😄 

Some repos have moved, some have deprecation notices in them, one is even outright gone.

Closes #10210.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by HonkingGoose** [comment]
Unsure if I need to add the `!` character in the PR title or not.
Conventional Commit style seems to dictate that you use a `!` after `type`/`scope` to denote breaking changes.

Feel free to remove the `!` character from the PR title if this will lead to problems. 😉

**Comment by rarkins** [comment]
> Unsure if I need to add the `!` character in the PR title or not.
> Conventional Commit style seems to dictate that you use a `!` after `type`/`scope` to denote breaking changes.
> 
> Feel free to remove the `!` character from the PR title if this will lead to problems. 😉

Can you check what semantic-release supports?

**Comment by HonkingGoose** [comment]
> Can you check what semantic-release supports?

Will do!

**Comment by HonkingGoose** [comment]
Quote from https://github.com/semantic-release/semantic-release/discussions/1796#discussioncomment-364000: 

> based on this log output, it looks like you are including the `BREAKING CHANGE:` in the subject of the commit. the default convention used by semantic-release expects this to be in the commit body, unlike `feat`, `fix`, etc because any commit type can include a breaking change

Also there's this from `semantic-release` README (https://github.com/semantic-release/semantic-release#commit-message-format):

> Here is an example of the release type that will be done based on a commit messages:
> 
> | Commit message                                                                                                                                                                                   | Release type               |
> | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------- |
> | `fix(pencil): stop graphite breaking when too much pressure applied`                                                                                                                             | Patch Release              |
> | `feat(pencil): add 'graphiteWidth' option`                                                                                                                                                       | ~~Minor~~ Feature Release  |
> | `perf(pencil): remove graphiteWidth option`<br><br>`BREAKING CHANGE: The graphiteWidth option has been removed.`<br>`The default graphite width of 10mm is always used for performance reasons.` | ~~Major~~ Breaking Release |

I think the only thing `semantic-release` supports is putting `BREAKING CHANGE: ` in the commit body.

**Comment by travi** [comment]
> I think the only thing `semantic-release` supports is putting `BREAKING CHANGE:` in the commit body.

this is true because we [still use the angular convention by default](https://github.com/semantic-release/semantic-release#commit-message-format) today. we intend to switch that default to conventional commits in the future, and that change can be tracked [here](https://github.com/semantic-release/semantic-release/pull/1836). until then, you could [change your convention preset](https://github.com/semantic-release/commit-analyzer#options), if you'd like.

**Comment by HonkingGoose** [comment]
It seems we're already using the conventional commit preset from `@semantic-release/commit-analyzer`:

https://github.com/renovatebot/renovate/blob/03a04dee445c3d81e38abac5be039a685f2b5016/.releaserc#L32

Link to file on current `main`: https://github.com/renovatebot/renovate/blob/main/.releaserc

What do you think @travi? Would it be safe to use a PR title like this: `feat(config)!: improve monorepo names` so with the `!` to signify breaking change?

I'll keep things as they are until we know for sure how we've configured `semantic-release` on our end. 😄

**Comment by travi** [comment]
> Would it be safe to use a PR title like this: `feat(config)!: improve monorepo names` so with the `!` to signify breaking change?

assuming you are mentioning the PR title because the PR will be squashed (and since the PR title is used as the default commit message when squashing), i would expect that to result in being treated as a breaking change, as you expect. it is important to keep in mind that `semantic-release` does not care about the actual PR, but only the resulting commits.

**Comment by HonkingGoose** [comment]
> assuming you are mentioning the PR title because the PR will be squashed

All our PRs get squashed into one commit, with commits from the PR branch going in the commit body, and the PR title as commit message.

> i would expect that to result in being treated as a breaking change, as you expect

Cool!

> it is important to keep in mind that `semantic-release` does not care about the actual PR, but only the resulting commits.

Will do!

**Comment by HonkingGoose** [comment]
@rarkins I've left some comments in the code, can you maybe review the PR and tell me what you think?

**Comment by HonkingGoose** [comment]
Summary: I'm bad at dealing with merge conflicts, and need a way to avoid them on this PR. 😉 

---

I'm having trouble fixing the merge conflict(s) on this PR every time I merge the current `main` branch.

The basic problem:

1. HonkingGoose creates breaking change in the monorepo config file
1. Renovate maintainers merge in new lines in monorepo config file to the `main` branch
1. HonkingGoose updates this PR to be current with `main`: _merge conflict_ ☠️ 
1. Manually fix merge conflict, trying not to mess up the lines that are good as-is

I'm bad at dealing with merge conflicts, and rather avoid them completely. 🙈 
Because my change is a breaking one, it's expected that things break regularly... 😄 

I think the root cause is that I'm trying to keep up-to-date with a branch that's moving under me. Which will always end badly... 😄 

I think we have the following options to get out of this mess:

- Abandon PR and try again once we're gearing up for a breaking release anyways **and** are willing to hold off adding new monorepo lines, so I can focus on cleaning up the existing lines
- Retry this PR on current `main`, with the promise from Renovate maintainers to stop merging new monorepo lines (probably a non-starter)
- Renovate team doesn't really care about this: close PR, and stop working on this problem

Or maybe one of you has a great idea? I'm open to suggestions! 😄

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:40]:
```suggestion
  'react-apollo': 'https://github.com/apollographql/react-apollo',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:48]:
```suggestion
  'telus-tds-core': 'https://github.com/telus/tds-core',
```

But do all packages point to the new URL already, or should we keep both?

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:55]:
```suggestion
  angularjs: 'https://github.com/angular/angular.js',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:57]:
```suggestion
  'angular-cli': 'https://github.com/angular/angular-cli',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:60]:
```suggestion
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:91]:
```suggestion
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:93]:
```suggestion
  javahamcrest: 'https://github.com/hamcrest/JavaHamcrest',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:97]:
```suggestion
  'infrastructure-ui': 'https://github.com/instructure/instructure-ui',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:104]:
```suggestion
  linguijs: 'https://github.com/lingui/js-lingui',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:110]:
```suggestion
    'https://github.com/material-components/material-components-web',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:112]:
```suggestion
  'material-ui': 'https://github.com/mui-org/material-ui',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:116]:
```suggestion
    'https://github.com/mozilla-neutrino/neutrino-dev',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:149]:
```suggestion
  'stryker-js': 'https://github.com/stryker-mutator/stryker-js',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:152]:
```suggestion
    'https://github.com/domaindrivendev/Swashbuckle.AspNetCore',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:164]:
```suggestion
  clarity: ['^@cds/', '^@clr/'],
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts]: @HonkingGoose FYI I think you can "batch" update commit suggestions using GitHub's web UI in case that's easier than doing them one by one

Comment (HonkingGoose) [reply lib/config/presets/internal/monorepo.ts]: > FYI I think you can "batch" update commit suggestions using GitHub's web UI in case that's easier than doing them one by one

Thank you for the tip! I know I can batch commits together, but that loses the context of the change.

I have no clue how to fix up this merge conflict, so I think the only way for _me_ to proceed is to re-do the work in full on the `main` branch?

Do you have any tips on how to fix this branch?

Comment (rarkins) [reply lib/config/presets/internal/monorepo.ts]: I think I've resolved it - please double check

File: lib/config/presets/internal/monorepo.ts
Comment (HonkingGoose) [reply lib/config/presets/internal/monorepo.ts:48]: I don't know about that, probably should stay safe and keep both.

**Review by HonkingGoose — COMMENTED** [review]
Thanks for resolving the merge conflict! 😄 

We still have some comments that need review/removal though. 😉

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:136]:
```suggestion
  feign: 'https://github.com/OpenFeign/feign',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:139]:
```suggestion
    'https://github.com/open-telemetry/opentelemetry-dotnet',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:140]:
```suggestion
  picassojs: 'https://github.com/qlik-oss/picasso.js',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:147]:
```suggestion
  'react-router': 'https://github.com/ReactTraining/react-router',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:152]:
```suggestion
  'sentry-javascript': 'https://github.com/getsentry/sentry-javascript',
```

File: lib/config/presets/internal/monorepo.ts
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:163]:
```suggestion
  typefaces: 'https://github.com/KyleAMathews/typefaces',
```

**Review by rarkins — CHANGES_REQUESTED** [review]
I think in general we can keep deprecated ones for quite some time - as long as people may have PRs updating to versions which still reference the old repo

**Comment by HonkingGoose** [comment]
@rarkins maybe this should go in the `v26` roll-up PR/branch as well? Let me know if I need to do anything... 😉

**Comment by rarkins** [comment]
@HonkingGoose I'll merge it into v26 if you're happy with it

**Review by rarkins — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Not sure if there is a pending release for this, but it's breaking updates in v26.1.2

**Comment by HonkingGoose** [comment]
@tomkerkhove we put these changes into a `v26` release as changing the monorepo names change the branch names as well, which can cause breakage.

Can you provide more details on how things are going wrong in `v26.1.2`?

Pinging @rarkins to listen to this conversation as well.

**Comment by rarkins** [comment]
The app is not yet updated to v26. @tomkerkhove can you provide more details?

**Comment by tomkerkhove** [comment]
We are getting the following:
```
INFO: Throwing preset error (repository=iot-platform/exedra-asset-management)
       "validationError": "Cannot find preset's package (group:SwashbuckleMonorepo). Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself."
```

Where we had to change this in our config:
```diff "extends": [
    "config:base",
    ":gitSignOff",
    "group:azure azure-libraries-for-netMonorepo",
    "group:azure azure-sdk-for-netMonorepo",
    "group:azure azure-storage-netMonorepo",
-    "group:swashbuckle-aspnetcoreMonorepo"
+    "group:SwashbuckleMonorepo"
  ]
```

Same for ASP.NET Extensions:
```
 INFO: Throwing preset error (repository=iot-platform/exedra-asset-management)
       "validationError": "Cannot find preset's package (monorepo:aspnet Extensions). Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself."
 INFO: Repository has invalid config (repository=iot-platform/exedra-asset-management)
       "error": {
         "validationError": "Cannot find preset's package (monorepo:aspnet Extensions). Note: this is a *nested* preset so please contact the preset author if you are unable to fix it yourself.",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at resolveConfigPresets (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/config/presets/index.js:265:35)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async resolveConfigPresets (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/config/presets/index.js:311:38)\n    at async Object.resolveConfigPresets (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/config/presets/index.js:289:38)\n    at async Object.mergeRenovateConfig (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/workers/repository/init/merge.js:210:50)\n    at async Object.getRepoConfig (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/workers/repository/init/config.js:12:14)\n    at async Object.initRepo (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/workers/repository/init/index.js:26:14)\n    at async Object.renovateRepository (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/workers/repository/index.js:64:18)\n    at async Object.start (/home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/workers/global/index.js:109:13)\n    at async /home/vsts/.npm/_npx/1776/lib/node_modules/renovate/dist/renovate.js:33:24"
       }
```

Note that these are written as INFO so our pipeline passes, while Renovate couldn't do anything

This is OK btw, just want to make sure you are aware of it.

*(this is a self-hosted scenario)*

**Comment by rarkins** [comment]
We put it in a major release intentionally, although I guess we failed to declare it as breaking:

image

@viceice @HonkingGoose I think we should add *migration* code to migrate all the old names to new names automatically. Otherwise we might see a lot of this suddenly in the app once it goes live

**Comment by tomkerkhove** [comment]
That would be nice to have indeed, thank you for that!

But you are correct - We just use the latest version so this is OK but just wanted to let you know in case you were not aware of it.

Maybe the error can have a link to the notes or so?

**Comment by rarkins** [comment]
Definitely thank you for reporting it as I wasn't aware yet. I think best we do automigration internally so that others like you don't even need to think about it.

**Comment by tomkerkhove** [comment]
Thanks a ton!

**Comment by viceice** [comment]
Yes. 🤦‍♂️

**Pull request merged**
